As with SciPy, NumPy abuses specifications of *build requirements* as a means
to control the environment under which wheels are built for distribution.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,7 +2,7 @@
 build-backend = "mesonpy"
 requires = [
     "Cython>=0.29.34,<3.1",
-    "meson-python>=0.15.0,<0.16.0",
+    "meson-python>=0.15.0",
 ]
 
 [project]
